Settings-default.toml

```diff
@@ -29,6 +28,7 @@ database = "geoengine"
 schema = "public"
 user = "geoengine"
 password = "geoengine"
+clear_database_on_start = false # set to `true` to start with a fresh database every time. This has to be set at the very first startup with an empty database.
```

Comment:
we should log if this is set to true but we ignore it for security reasons

Reply:
I do log it at info level on startup

Reply:
with clear = true on a database with clear = true
![image](https://github.com/geo-engine/geoengine/assets/10283892/3f1e2753-b223-4d69-99c6-d56fc39649a1)

with clear = true on a datbase with clear = false
![image](https://github.com/geo-engine/geoengine/assets/10283892/6708e3d0-e05e-4d23-a64b-00a4ccf44729)

Reply:
1. it should be an error if `clear = false` at the first start and `clear = true` at a consecutive one.
2. The error does not state what went wrong.

Reply:
1. that's what I meant and what I did
2. it does state that ClearDatabaseOnStartupNotAllowed. I don't actually know why it doesn't print my snafu display message that explain it in more detail. But that's more of a general error handling issue, than an issue with my PR.

Reply:
I tried this:

1. Start with `clear = true`
2. Start with `clear = false`
3. Start with `clear = true`

no error

Reply:
IMHO it is correct that there is no error. The first time you start the engine you determine whether clearOnStart is possible, and this is saved in the database.

If you then decide to not clear it the next time, it should be fine, everything survives.

If you then start again with clear, it should clear, because it is allowed in the database.

At least that's my thinking. It would protect production databases that initially set clear to true, but allows us to ad-hoc decide whether we want to truncate the data during development and testing.